### DISCUSSION TITLE
TEAM Terraform Provider

### DISCUSSION THREAD

**Discussion by brittandeyoung** [aws-samples/iam-identity-center-team#131] [Ideas]
Having the ability to programmatically configure TEAM settings, approvers, and eligibilities would be highly beneficial. We currently automate the deployment of our multi account structure, but are having to manually create approvers and eligibilities for these new accounts as they spin up. 

I have experience Writing terraform providers and am a core contributor to the AWS provider for terraform. I Would like to write a provider for TEAM. I am running into issues when it comes to authorizing my request with the appsync graph api. I am able to create a new client with a client ID and secret, then use these to get a token. This token is allowing me to perform all graphql query operations, but I am receiving a `"Not Authorized to access createSettings on type Mutation"` error when attempting to make mutations. 

It looks like authorization is currently based off of user, does the project need a new feature to support machine authentication against the api?

**Comment by brittandeyoung** [comment]
I am now seeing that there is an open issue for adding API support: #33 

One question I have about the provider is would `awsteam` be the appropriate provider name?

I currently have my WIP code here: https://github.com/brittandeyoung/terraform-provider-awsteam
and could likely publish the provider shortly after API support is added.

**Comment by brittandeyoung** [comment]
The terraform provider has been published here: https://registry.terraform.io/providers/brittandeyoung/awsteam/latest/docs

**Reply by tawoyinfa** [reply]
Fantastic contribution @brittandeyoung

**Comment by tawoyinfa** [comment]
@brittandeyoung  do you reckon you can link/add the terraform provider (examples/schemas etc) to the main TEAM documentation ? I think it will help it to gain traction

**Reply by brittandeyoung** [reply]
The provider docs are currently linked to here: https://aws-samples.github.io/iam-identity-center-team/docs/deployment/configuration/cognito_machine_auth.html

does it need to be in a more prominent spot?

**Comment by brittandeyoung** [comment]
@tawoyinfa I have raised a PR to help improve the visibility of the terraform provider #184 .

This is in response to an issue that was raised asking for automated configuration methods. This will hopefully make it easier to find.
